### PR TITLE
validateRegisterForm

### PR DESCRIPTION
**Pull request by Viktoriia17051991**

### CODE REVIEW

File: src/validateRegisterForm.test.js [hunk 1]
```diff
@@ -26,5 +26,57 @@ describe(`Function 'validateRegisterForm':`, () => {
     expect(invalidPassword.message).toBe('Password is invalid.');
   });
 
-  // write more tests here
+  it(`should return error for valid email and password less 
+  than 8 characters`, () => {
+    const invalidPassword = validateRegisterForm('test@mail.com', 'P@ss1');
```
Comment (4ksandrey) [inline src/validateRegisterForm.test.js:31]: what about boundary values?) 8-16 contains at least 1 digit, 1 special character, and 1 uppercase letter. So use boundary values techniques for these requirements and cover them by automated tests: 7, 8, 16, 17, 10 without capital, 10 without spec char, 10 without digit

File: src/validateRegisterForm.test.js [hunk 2]
```diff
@@ -26,5 +26,57 @@ describe(`Function 'validateRegisterForm':`, () => {
...
+   than 16 characters`, () => {
+    const invalidPassword = validateRegisterForm('test@mail.com',
+      'P@ssword123456789!');
+
+    expect(invalidPassword.code).toBe(422);
+    expect(invalidPassword.message).toBe('Password is invalid.');
+  });
+
+  it(`should return error for invalid email without '@' symbol`, () => {
+    const invalidEmail = validateRegisterForm('testmail.com', 'P@ssword1!');
```
Comment (4ksandrey) [inline src/validateRegisterForm.test.js:47]: If entered password and email are invalid object contains status code 500 and message Password and email are invalid. This requirement isn't covered

**Review by 4ksandrey — CHANGES_REQUESTED** [review]

File: src/validateRegisterForm.test.js [hunk 3]
```diff
@@ -26,5 +26,119 @@ describe(`Function 'validateRegisterForm':`, () => {
...
+    const isValid = validateRegisterForm('test@mail.com',
+      'P@ssword1234567');
+
+    expect(isValid.code).toBe(200);
+    expect(isValid.message).toBe('Email and password are valid.');
+  });
+
+  it(`should return error for valid email and password of length 17`,
+    () => {
+      const invalidPassword = validateRegisterForm('test@mail.com',
```
Comment (PolinaRomanenko) [inline src/validateRegisterForm.test.js:57]: The test description says it should return an error, but the test expects a success code (200). Update the test description or the expected code accordingly.

File: src/validateRegisterForm.test.js [hunk 4]
```diff
@@ -26,5 +26,119 @@ describe(`Function 'validateRegisterForm':`, () => {
...
+
+      expect(invalidEmail.code).toBe(422);
+      expect(invalidEmail.message).toBe('Email is invalid.');
+    });
+
+  it(`should return error for invalid email with top-level
+   domain starting with a dot`, () => {
+    const invalidEmail = validateRegisterForm('test@mail.com.', 'P@ssword1!');
+
+    expect(invalidEmail.code).toBe(200);
```
Comment (PolinaRomanenko) [inline src/validateRegisterForm.test.js:131]: The test description says it should return an error, but the test expects a success code (200). Update the test description or the expected code accordingly.

File: src/validateRegisterForm.test.js [hunk 5]
```diff
@@ -26,5 +26,119 @@ describe(`Function 'validateRegisterForm':`, () => {
...
+    const invalidEmail = validateRegisterForm('.test@mail.com', 'P@ssword1!');
+
+    expect(invalidEmail.code).toBe(422);
+    expect(invalidEmail.message).toBe('Email is invalid.');
+  });
+
+  it(`should return error for invalid email ending with '.'`, () => {
+    const invalidEmail = validateRegisterForm('test@mail.com.', 'P@ssword1!');
+
+    expect(invalidEmail.code).toBe(200);
```
Comment (PolinaRomanenko) [inline src/validateRegisterForm.test.js:114]: The test description says it should return an error, but the test expects a success code (200). Update the test description or the expected code accordingly.

**Review by PolinaRomanenko — CHANGES_REQUESTED** [review]
fix the comments, please

File: src/validateRegisterForm.test.js [hunk 6]
```diff
@@ -26,5 +26,119 @@ describe(`Function 'validateRegisterForm':`, () => {
...
+   and password of length 16`, () => {
+    const isValid = validateRegisterForm('test@mail.com',
+      'P@ssword1234567');
+
+    expect(isValid.code).toBe(200);
+    expect(isValid.message).toBe('Email and password are valid.');
+  });
+
+  it(`should return error for valid email and password less 
+  than 17 characters`, () => {
```
Comment (irini132) [inline src/validateRegisterForm.test.js:56]: This test case is misleading as it states 'should return error for valid email and password less than 17 characters', but the password 'P@ss1' is only 5 characters long.

File: src/validateRegisterForm.test.js [hunk 7]
```diff
@@ -26,5 +26,119 @@ describe(`Function 'validateRegisterForm':`, () => {
...
+    starting with a dot`, () => {
+    const invalidEmail = validateRegisterForm('example@mail.com.',
+      'P@ssword1!');
+
+    expect(invalidEmail.code).toBe(422);
+    expect(invalidEmail.message).toBe('Email is invalid.');
+  });
+
+  it(`should return error for invalid email and password`, () => {
+    const invalidEmailAndPassword = validateRegisterForm('invalidemail',
```
Comment (irini132) [inline src/validateRegisterForm.test.js:136]: In the test case 'should return error for invalid email and password', the email and password used do not clearly represent invalid cases. It would be better to use clearly invalid examples (e.g. 'email_without_at_symbol' and 'passwordWithoutSpecialCharOrNumber') to enhance readability and clarity.

File: src/validateRegisterForm.test.js [hunk 8]
```diff
@@ -26,5 +26,119 @@ describe(`Function 'validateRegisterForm':`, () => {
...
+  });
+
+  it(`should return error for invalid email starting with '.'`, () => {
+    const invalidEmail = validateRegisterForm('.test@mail.com', 'P@ssword1!');
+
+    expect(invalidEmail.code).toBe(422);
+    expect(invalidEmail.message).toBe('Email is invalid.');
+  });
+
+  it(`should return error for email not ending with '.'`, () => {
```
Comment (irini132) [inline src/validateRegisterForm.test.js:110]: This test case is misleading as it states 'should return error for email not ending with '.'', but the email 'test@mail.com' does end with '.'. The expectations then contradict the description as they check for a non-error code and message.

**Review by irini132 — CHANGES_REQUESTED** [review]
Check the comment please

**Review by VikVolSen — APPROVED** [review]
Good job!
